cni-plugin/pkg/plugin/plugin.go

```diff
@@ -104,6 +107,41 @@ func testConnection() error {
 	return nil
 }
 
+func isEndpointReady(readyEndpoint string, timeout time.Duration) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	c := &http.Client{}
+	req, err := http.NewRequest(http.MethodGet, readyEndpoint, nil)
+	if err != nil {
+		return false, err
+	}
+	req = req.WithContext(ctx)
+	resp, err := c.Do(req)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode < 200 || resp.StatusCode >= 400 {
+		return false, fmt.Errorf("Endpoint is not ready, response code returned:%d", resp.StatusCode)
+	}
+	return true, nil
+}
+
+func pollReadinessEndpoint(endpoint string, interval, timeout time.Duration) error {
```

Comment:
nit: can we rename the function to `pollEndpointReadiness()`? (disclaimer: English is not my 1st language, but I think that sounds more natural)

Reply:
Done.